pkg/core/mempool/mem_pool.go

```diff
@@ -109,22 +109,16 @@ func (mp *Pool) containsKey(hash util.Uint256) bool {
 // tryAddSendersFee tries to add system fee and network fee to the total sender`s fee in mempool
 // and returns false if sender has not enough GAS to pay
 func (mp *Pool) tryAddSendersFee(tx *transaction.Transaction, feer Feer) bool {
-	if !mp.checkBalanceAndUpdate(tx, feer) {
-		return false
-	}
-	mp.addSendersFee(tx)
-	return true
-}
-
-// checkBalanceAndUpdate returns true in case when sender has enough GAS to pay for
-// the transaction and sets sender's balance value in mempool in case if it was not set
-func (mp *Pool) checkBalanceAndUpdate(tx *transaction.Transaction, feer Feer) bool {
 	senderFee, ok := mp.fees[tx.Sender]
 	if !ok {
 		senderFee.balance = feer.GetUtilityTokenBalance(tx.Sender)
 		mp.fees[tx.Sender] = senderFee
 	}
```

Comment:
It duplicates `addSendersFee`, I think we can deduplicate them.

Reply:
Finally managed to join these methods with a help of `needCheck` flag.